src/index.js

```diff
@@ -224,6 +224,30 @@ const userEvent = {
       }
     }
     element.addEventListener("blur", fireChangeEvent);
+  },
+
+  tab({ shift = false } = {}) {
+    const focusableElements = document.querySelectorAll(
+      "input, button, select, textarea, a[href], [tabindex]"
+    );
+    const list = Array.prototype.filter.call(focusableElements, function (item) {
+      return item.getAttribute("tabindex") !== "-1";
+    }).sort((a, b) => {
+      const tabIndexA = a.getAttribute('tabindex');
+      const tabIndexB = b.getAttribute('tabindex');
+      return tabIndexA < tabIndexB ? -1 : tabIndexA > tabIndexB ? 1 : 0;
+    });
+    const index = list.indexOf(document.activeElement);
+
+    let nextIndex = shift ? index - 1 : index + 1;
+    let defaultIndex = shift ? list.length - 1 : 0;
+
+    const next = list[nextIndex] || list[defaultIndex];
+
+    if (next.getAttribute("tabindex") === null) {
+      next.setAttribute("tabindex", "0"); // jsdom requires tabIndex=0 for an item to become 'document.activeElement' (the browser does not)
```

Comment:
Should we unset this after `next.focus()`? It would be good if this function could be as pure as possible.

Reply:
@kentcdodds good call, i'd thought about this but was probably driving and forgot to add it. lol

done